wordpress-mobile/WordPress-iOS
# Add JetpackSocialService to fetch auto-sharing limit

Refs #20806 
Depends on #20946

Adds a service class to fetch & update `PublicizeInfo` and saves them into Core Data. Some notes:
- The new information will always be prioritized. Meaning, if there's pre-existing data, it will be updated with the new data. If the new information is `nil`, any pre-existing data will be deleted.
- Added a new value-type model, `PublicizeInfo.SharingLimit`, which contains only the `sharesRemaining` and `shareLimit`. This is so the completion block doesn't have to deal with a managed object (i.e. `PublicizeInfo`).

## To test

- Added some unit tests. Please run them and ensure that everything is 🟢.
- There are no user-facing changes yet as of now.

## Regression Notes
1. Potential unintended areas of impact
Should be none. Method performs some light Core Data operation but not called yet.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Added unit tests.

4. What automated tests I added (or what prevented me from doing so)
Added some unit tests to cover the new class.

PR submission checklist:


UI Changes testing checklist:

N/A. There are no user-facing changes in this PR.
